tests/test_tasks/test_task.py

```diff
@@ -1,77 +1,42 @@
-        retval = openml.tasks.get_task(1)
-        self.assertEqual(patch.call_count, 1)
-        self.assertIsInstance(retval, openml.OpenMLTask)
-        self.assertEqual(retval.class_labels, 'LA')
-
-    def test_get_X_and_Y(self):
-        # Classification task
-        task = openml.tasks.get_task(1)
-        X, Y = task.get_X_and_y()
-        self.assertEqual((898, 38), X.shape)
-        self.assertIsInstance(X, np.ndarray)
-        self.assertEqual((898, ), Y.shape)
-        self.assertIsInstance(Y, np.ndarray)
-        self.assertEqual(Y.dtype, int)
-
-        # Regression task
-        task = openml.tasks.get_task(631)
-        X, Y = task.get_X_and_y()
-        self.assertEqual((52, 2), X.shape)
-        self.assertIsInstance(X, np.ndarray)
-        self.assertEqual((52,), Y.shape)
-        self.assertIsInstance(Y, np.ndarray)
-        self.assertEqual(Y.dtype, float)
-
-    def test_tagging(self):
-        task = openml.tasks.get_task(1)
-        tag = "testing_tag_{}_{}".format(self.id(), time())
-        task_list = openml.tasks.list_tasks(tag=tag)
-        self.assertEqual(len(task_list), 0)
-        task.push_tag(tag)
-        task_list = openml.tasks.list_tasks(tag=tag)
-        self.assertEqual(len(task_list), 1)
-        self.assertIn(1, task_list)
-        task.remove_tag(tag)
-        task_list = openml.tasks.list_tasks(tag=tag)
-        self.assertEqual(len(task_list), 0)
-
-    def test_get_train_and_test_split_indices(self):
-        openml.config.cache_directory = self.static_cache_dir
-        task = openml.tasks.get_task(1882)
-        train_indices, test_indices = task.get_train_test_split_indices(0, 0)
-        self.assertEqual(16, train_indices[0])
-        self.assertEqual(395, train_indices[-1])
-        self.assertEqual(412, test_indices[0])
-        self.assertEqual(364, test_indices[-1])
-        train_indices, test_indices = task.get_train_test_split_indices(2, 2)
-        self.assertEqual(237, train_indices[0])
-        self.assertEqual(681, train_indices[-1])
-        self.assertEqual(583, test_indices[0])
-        self.assertEqual(24, test_indices[-1])
-        self.assertRaisesRegexp(ValueError, "Fold 10 not known",
-                                task.get_train_test_split_indices, 10, 0)
-        self.assertRaisesRegexp(ValueError, "Repeat 10 not known",
-                                task.get_train_test_split_indices, 0, 10)
+    # task id will be set from the
+    # extending classes
+
+    def setUp(self):
+
+        super(OpenMLTaskTest, self).setUp()
+        self.task_id = 11
+        self.estimation_procedure = 23
+
+    @classmethod
+    def setUpClass(cls):
+        if cls is OpenMLTaskTest:
+            raise unittest.SkipTest(
+                "Skip OpenMLTaskTest tests,"
+                " it's a base class"
+            )
+        super(OpenMLTaskTest, cls).setUpClass()
+
+    def test_download_task(self):
+
+        openml.tasks.get_task(self.task_id)
```

Comment:
Could you please add a few asserts here to check that the downloaded content is as expected?

Reply:
Yes, I can.

Reply:
Now that I am revisiting this comment, adding asserts here is not possible, since this method will be called only on the classes that extend the base class and that information is task specific.
What I can do is override this method on the extending classes and perform the check according to the tasks.

Reply:
That would be great.